src/integrationTest/java/org/opensearch/test/framework/certificate/TestCertificates.java

```diff
@@ -32,46 +32,146 @@
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static org.opensearch.test.framework.certificate.PublicKeyUsage.CLIENT_AUTH;
+import static org.opensearch.test.framework.certificate.PublicKeyUsage.CRL_SIGN;
+import static org.opensearch.test.framework.certificate.PublicKeyUsage.DIGITAL_SIGNATURE;
+import static org.opensearch.test.framework.certificate.PublicKeyUsage.KEY_CERT_SIGN;
+import static org.opensearch.test.framework.certificate.PublicKeyUsage.KEY_ENCIPHERMENT;
+import static org.opensearch.test.framework.certificate.PublicKeyUsage.NON_REPUDIATION;
+import static org.opensearch.test.framework.certificate.PublicKeyUsage.SERVER_AUTH;
 
 /**
-* Provides TLS certificates required in test cases.
-* WIP At the moment the certificates are hard coded. 
-* This will be replaced by classes
-* that can generate certificates on the fly.
+* It provides TLS certificates required in test cases. The certificates are generated during process of creation objects of the class.
+* The class exposes method which can be used to write certificates and private keys in temporally files.
 */
 public class TestCertificates {
 
+	public static final Integer MAX_NUMBER_OF_NODE_CERTIFICATES = 3;
+
+	private static final String CA_SUBJECT = "DC=com,DC=example,O=Example Com Inc.,OU=Example Com Inc. Root CA,CN=Example Com Inc. Root CA";
```

Comment:
In some of our tests I see that we have intermediate signing certificates: https://github.com/opensearch-project/security/blob/main/src/test/java/org/opensearch/security/ssl/SecuritySSLCertsInfoActionTests.java#L36

With subject `CN=Example Com Inc. Signing CA,OU=Example Com Inc. Signing CA,O=Example Com Inc.,DC=example,DC=com`.

Can we create intermediate signing CAs with this framework as well?

Reply:
To be honest, I did not test this, but I assume that it should be possible. The intermediate certificate is much like a root certificate (basic constrain CA:true, and proper key usage, etc.) but additionally signed with the private key associated with the parent certificate.